tests/test_opossum_file.py

```diff
@@ -1,24 +1,42 @@
 # SPDX-FileCopyrightText: 2023 TNG Technology Consulting GmbH <https://www.tngtech.com>
 #
 # SPDX-License-Identifier: Apache-2.0
-from opossum_lib.opossum_file import Resource
+from opossum_lib.opossum_file import Resource, ResourceType
 
 
-def test_resource_to_dict() -> None:
+def test_resource_to_dict_with_file_as_leaf() -> None:
     list_of_paths = [["A"], ["A", "B", "C"], ["A", "B"], ["A", "D"]]
-    resource = Resource()
+    resource = Resource(ResourceType.TOP_LEVEL)
 
     for path in list_of_paths:
-        resource.add_path(path)
+        resource.add_path(path, ResourceType.FILE)
```

Comment:
i'm surprised this is possible - "A" is a file but has children?

Reply:
Yes, at least kind of. When converting an SPDX document the relationships within the document are visualized in the file path tree in a way that the "left-hand-side" element of the relationship contains the relationship type which again contains the "right-hand-side" of the relationship. In general it is possible that an SPDX file is on the "left-hand-side" and I think it is confusing to give a file the `ResourceType.FOLDER`. 
Example for that:
![image](https://github.com/opossum-tool/opossum.lib.py/assets/50019307/c076eacc-344b-499a-bab4-3bb533cbb5a5)
In this case `LICENSE` is a license file but as it is the "left-hand-side" of the relationship it is marked as a folder.

Reply:
Interesting, thanks